spyder/plugins/explorer/widgets/fileassociations.py

```diff
@@ -0,0 +1,525 @@
+        self.button_cancel = self.button_box.button(QDialogButtonBox.Cancel)
+
+        # Widget setup
+        self.setWindowTitle(_('Applications dialog'))
+        self.edit_filter.setPlaceholderText(_('Type to filter by name...'))
+        self.list.setIconSize(QSize(16, 16))  # FIXME: Use metrics
+
+        # Layout
+        layout = QVBoxLayout()
+        layout.addWidget(self.label)
+        layout.addWidget(self.edit_filter)
+        layout.addWidget(self.list)
+        layout_browse = QHBoxLayout()
+        layout_browse.addWidget(self.button_browse)
+        layout_browse.addWidget(self.label_browse)
+        layout.addLayout(layout_browse)
+        layout.addSpacing(12)  # FIXME: Use metrics
+        layout.addWidget(self.button_box)
+        self.setLayout(layout)
+
+        # Signals
+        self.edit_filter.textChanged.connect(self.filter)
+        self.button_browse.clicked.connect(lambda x: self.browse())
+        self.button_ok.clicked.connect(self.accept)
+        self.button_cancel.clicked.connect(self.reject)
+
+        self.setup()
+
+    def setup(self, applications=None):
+        """Load installed applications."""
+        QApplication.setOverrideCursor(QCursor(Qt.WaitCursor))
+        self.list.clear()
+        if applications is None:
+            apps = get_installed_applications()
+        else:
+            apps = applications
+
+        for app in sorted(apps, key=lambda x: x.lower()):
+            fpath = apps[app]
+            icon = get_application_icon(fpath)
+            item = QListWidgetItem(icon, app)
+            item.setToolTip(fpath)
+            item.fpath = fpath
+            self.list.addItem(item)
+
+        # FIXME: Use metrics
+        self.list.setMinimumWidth(self.list.sizeHintForColumn(0) + 24)
+        QApplication.restoreOverrideCursor()
+
+    def browse(self, fpath=None):
+        """Prompt user to select an application not found on the list."""
+        app = None
+        item = None
+
+        if sys.platform == 'darwin':
+            if fpath is None:
+                basedir = '/Applications/'
+                filters = _('Applications (*.app)')
+                title = _('Select application')
+                fpath, __ = getopenfilename(self, title, basedir, filters)
+
+            if fpath and fpath.endswith('.app'):
+                app = os.path.basename(fpath).split('.app')[0]
+                for row in range(self.list.count()):
+                    item = self.list.item(row)
+                    if app == item.text() and fpath == item.fpath:
+                        break
+                else:
+                    item = None
+        elif os.name == 'nt':
+            if fpath is None:
+                basedir = 'C:\\'
+                filters = _('Applications (*.exe *.bat *.com)')
+                title = _('Select application')
+                fpath, __ = getopenfilename(self, title, basedir, filters)
+
+            if fpath and fpath.endswith(('.exe', '.bat', '.com')):
```

Comment:
We should also test for binary files here, to avoid false positives.

Or do a more fine-grained testing, perhaps? I mean `.exe` are binaries, `.bat` are plain text and I don't know about `.com`.

Reply:
Will do